src/styling.rs

```diff
@@ -0,0 +1,788 @@
+///   ف–و (U+0641..U+0648), ي (U+064A).
+/// - arabic-dotless: The dotless Arabic letter variants ٮ (U+066E), ٯ
+///   (U+066F), ڡ (U+06A1), ں (U+06BA)
+/// - digamma: The uppercase and lowercase digamma, Ϝ (U+03DC) and ϝ (U+03DD).
+/// - dotless: The dotless variants of the lowercase Latin letters i and j, ı
+///   (U+0131) and ȷ (U+0237).
+///
+/// Note that some styles support only a subset of a group. The characters each
+/// style supports are given in their documentation.
+///
+/// # Script style variants
+///
+/// There are two widely recognized variants of the script style: chancery and
+/// roundhand. They can be distinguished with variation sequences, by using the
+/// variation selectors U+FE00 and U+FE01 for chancery and roundhand
+/// respectively. These are specified in the [StandardizedVariants.txt] file
+/// from the Unicode Character Database.
+///
+/// Only the uppercase Latin letters are standardized variation sequences, but
+/// the [`Chancery`](MathStyle::Chancery) and
+/// [`Roundhand`](MathStyle::Roundhand) styles also support the lowercase Latin
+/// letters. In addition, the bold styles
+/// [`BoldChancery`](MathStyle::BoldChancery) and
+/// [`BoldRoundhand`](MathStyle::BoldRoundhand) are provided, which support
+/// both the uppercase and lowercase Latin letters despite not being specified
+/// as standardized variation sequences by Unicode.
+///
+/// # Shaping
+///
+/// The Arabic styles (including those from the
+/// [`Doublestruck`](MathStyle::Doublestruck) style) are not subject to
+/// shaping. However, [`Plain`](MathStyle::Plain) should still be shaped, as
+/// the characters are Arabic letters in the Arabic block (U+0600..U+06FF).
+///
+/// [StandardizedVariants.txt]: <https://www.unicode.org/Public/UNIDATA/StandardizedVariants.txt>
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, Default)]
+pub enum MathStyle {
+    /// Unstyled default. May be serif or sans-serif depending on the font.
+    #[default]
+    Plain,
+    /// Bold style. May be serif or sans-serif depending on the font.
+    ///
+    /// Supported characters: digits, latin, greek, digamma.
+    Bold,
+    /// Italic style. May be serif or sans-serif depending on the font.
+    ///
+    /// Supported characters: latin, greek, dotless, and the extra ħ (U+0127).
+    Italic,
+    /// Bold italic style. May be serif or sans-serif depending on the font.
+    ///
+    /// Supported characters: latin, greek.
+    BoldItalic,
+    /// Script style.
+    ///
+    /// Supported characters: latin.
+    Script,
+    /// Bold script style.
+    ///
+    /// Supported characters: latin.
+    BoldScript,
+    /// Fraktur style. Also known as black-letter style.
+    ///
+    /// Supported characters: latin.
+    Fraktur,
+    /// Bold fraktur style. Also known as bold black-letter style.
+    ///
+    /// Supported characters: latin.
+    BoldFraktur,
+    /// Sans-serif style.
+    ///
+    /// Supported characters: digits, latin.
+    Sansserif,
```

Comment:
```suggestion
    SansSerif,
```

Not very important, but writing "sans serif" as one word feels weird to me. Another alternative is simply `Sans`.

Reply:
`Sans` is very familiar since it's present in tons of font names (e.g. Comic Sans), so I'd slightly prefer that, especially given the length of `SansBoldItalic` vs `SansSerifBoldItalic`.

Reply:
IIRC my thought was that hyphens shouldn't be camel case otherwise it reads like two separate words "sans serif" instead of "sans-serif".

As with double-struck, despite being more verbose I think it'd be good to keep with the full name "sans-serif" used in Unicode.

Reply:
I think `SansSerif` is definitely more readable than `Sansserif`

Reply:
> IIRC my thought was that hyphens shouldn't be camel case otherwise it reads like two separate words "sans serif" instead of "sans-serif".

It *is* two separate words. Sans means "without", so it's just saying "without serifs".

Reply:
I think I meant that as two separate words they could then each read like a style modifier (e.g. Script and BoldScript): SansSerifBold is Sans-serif Bold not Sans (Serif Bold). Anyways I just made the change to SansSerif, and also to DoubleStruck.